integration_test/test_xgboost.py

```diff
@@ -1,51 +1,120 @@
 from sklearn.model_selection import train_test_split
+import numpy
 import xgboost as xgb
-import numpy as np
-
-iris = datasets.load_iris()
-X = iris.data
-y = iris.target
-num_class = 3
-X_train, X_test, Y_train, Y_test = train_test_split(X, y, test_size=0.2)
-D_train = xgb.DMatrix(X_train, label=Y_train)
-D_test = xgb.DMatrix(X_test, label=Y_test)
-
 
 POSSIBLE_PARAMETERS = {
   'eta': 10 ** random.uniform(-4, 1),
   'gamma': random.uniform(0, 4),
   'max_depth': random.randint(1, 5),
   'min_child_weight': random.uniform(0, 3),
-  'num_class': num_class,
   'lambda': random.uniform(1, 3),
   'alpha': 10 ** random.uniform(-4, 0),
-  'objective': random.choice(['multi:softmax', 'multi:softprob']),
   'tree_method': random.choice(['hist', 'exact', 'approx', 'auto']),
-  'eval_metric': ['mlogloss', 'merror'],
 }
 
-def _form_random_run_params():
-  random_subset_size = random.randint(1, len(POSSIBLE_PARAMETERS))
-  subset_keys = random.sample(POSSIBLE_PARAMETERS.keys(), random_subset_size)
-  if 'num_class' not in subset_keys:
-    subset_keys.append('num_class')
-  subset_param = {k: POSSIBLE_PARAMETERS[k] for k in subset_keys}
-  return dict(
-    params=subset_param,
-    dtrain=D_train,
-    evals=[(D_test, 'test0')],
-    num_boost_round=random.randint(3, 15),
-    verbose_eval=False,
-  )
+CLASSIFICATION_METRIC_NAMES = (
+  'accuracy',
+  'F1',
+  'precision',
+  'recall',
+)
+REGRESSION_METRIC_NAMES = (
+  'mean absolute error',
+  'mean squared error',
+)
+
+def _create_random_dataset(task='binary'):
+  if task == 'binary':
+    n_samples = random.randint(180, 300)
+    n_features = random.randint(5, 25)
+    n_classes = 2
+
+    return datasets.make_classification(
+      n_samples=n_samples,
+      n_features=n_features,
+      n_classes=n_classes,
+    )
+  elif task == 'multiclass':
+    n_samples = random.randint(180, 300)
+    n_classes = random.randint(3, 8)
+    n_informative = random.randint(2 * n_classes, 20)
+    n_features = random.randint(n_informative + 2, 40)
+
+    return datasets.make_classification(
+      n_samples=n_samples,
+      n_features=n_features,
+      n_informative=n_informative,
+      n_classes=n_classes,
+    )
+  else:
+    n_samples = random.randint(200, 500)
+    n_features = random.randint(50, 100)
+    n_informative = random.randint(10, n_features - 2)
+
+    return datasets.make_regression(
+      n_samples=n_samples,
+      n_features=n_features,
+      n_informative=n_informative,
+      noise=0.2,
+    )
+
+def _create_random_metric_objective(task='binary'):
+  if task == 'binary':
+    return {
+      'objective': random.choice(['binary:logistic', 'binary:hinge', 'binary:logitraw']),
+      'eval_metric': ['logloss', 'aucpr', 'error'],
+    }
+  elif task == 'multiclass':
+    return {
+      'objective': random.choice(['multi:softmax', 'multi:softprob']),
+      'eval_metric': ['mlogloss', 'merror'],
+    }
+  else:
+    return {
+      'objective': random.choice(['reg:squarederror']),
```

Comment:
If we're considering all the learning objectives for binary and multi class classification, might as well do the same for regression: 

`['reg:squarederror', 'reg:squaredlogerror', 'reg:logistic', 'reg:pseudohubererror', 'reg:gamma', 'reg:tweedie']`

(not sure if all of these are compatible with standard floating point data)

Reply:
The problem with `squaredlogerror` or `logistic` error is that they require the y to be of certain value ranges. Didn't have a quick way of dealing with it using the `make_regression` function. 

The rest should be ok? I guess

Reply:
hmm `gamma` and `tweedie` all requires y > 0.

Reply:
hmm okay never mind just reg:squarederror is fine then